### PR TITLE
quic: improve coverage

### PR DESCRIPTION
**Pull request by alyssawilk**
Arguably I should adjust this up but given recent CI failures I'll leave for now.

Risk Level: n/a (test only)
Testing: yes
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Review by danzh2010 — APPROVED** [review]

File: source/common/quic/client_connection_factory_impl.cc [hunk 1]
```diff
@@ -19,7 +19,6 @@ getContext(Network::TransportSocketFactory& transport_socket_factory) {
   auto* quic_socket_factory =
       dynamic_cast<QuicClientTransportSocketFactory*>(&transport_socket_factory);
   ASSERT(quic_socket_factory != nullptr);
-  ASSERT(quic_socket_factory->sslCtx() != nullptr);
```
Comment (antoniovicente) [inline source/common/quic/client_connection_factory_impl.cc:22]: If this ASSERT were to fail the following code would crash.  If we remove the ASSERT, we should also fix this code so it doesn't crash.

https://github.com/envoyproxy/envoy/blob/1ea2aa686e096a64be4683aa7d72350bcb60b9d2/source/common/quic/envoy_quic_proof_verifier.cc#L35

File: source/common/quic/quic_filter_manager_connection_impl.h [hunk 2]
```diff
@@ -77,7 +77,7 @@ class QuicFilterManagerConnectionImpl : public Network::ConnectionImplBase,
   absl::optional<Network::Connection::UnixDomainSocketPeerCredentials>
   unixSocketPeerCredentials() const override {
     // Unix domain socket is not supported.
-    NOT_REACHED_GCOVR_EXCL_LINE;
+    return absl::nullopt;
```
Comment (antoniovicente) [inline source/common/quic/quic_filter_manager_connection_impl.h:80]: I'm having trouble finding non-test uses of unixSocketPeerCredentials().

What am I missing?  Who depends on this API?

File: source/common/quic/client_connection_factory_impl.cc (hunk 1 above)
Comment (antoniovicente) [inline source/common/quic/client_connection_factory_impl.cc:22]: What tests do we have for getContext with sslCtx returning nullptr?

**Review by antoniovicente — COMMENTED** [review]
Thanks for the test coverage improvements.

Given that we are removing ASSERTs, I think we should add some nullptr checks to avoid crashes in EnvoyQuicProofVerifier::VerifyCertChain

File: source/common/quic/client_connection_factory_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/quic/client_connection_factory_impl.cc:22]: This is already removed over here https://github.com/envoyproxy/envoy/pull/16462
just hadn't merged it in.
That PR came with tests and error handling for SDS lazy loading of secrets :-)

File: source/common/quic/quic_filter_manager_connection_impl.h (hunk 2 above)
Comment (alyssawilk) [reply source/common/quic/quic_filter_manager_connection_impl.h:80]: Not sure. I'd be happy to remove as a follow-up if we think it's doable.
I wonder if it's used internally by whoever @snowp worked for at the time?

File: source/common/quic/client_connection_factory_impl.cc (hunk 1 above)
Comment (antoniovicente) [reply source/common/quic/client_connection_factory_impl.cc:22]: nit: Could we add ASSERT(context_ != nullptr) to the EnvoyQuicProofVerifier constructor as a sanity check?

https://github.com/envoyproxy/envoy/blob/main/source/common/quic/envoy_quic_proof_verifier.h#L15

**Review by antoniovicente — APPROVED** [review]

File: source/common/quic/client_connection_factory_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/quic/client_connection_factory_impl.cc:22]: added in https://github.com/envoyproxy/envoy/pull/16466 thanks

**Comment by wrowe** [comment]
This has broken four CI pipelines... better to revert for now?

**Comment by alyssawilk** [comment]
https://github.com/envoyproxy/envoy/pull/16597 should fix

**Pull request merged**